mvpa2/algorithms/searchlight_hyperalignment.py

```diff
@@ -119,6 +120,9 @@ def _call(self, ds):
             seed_index = None
         # Voxel selection within Searchlight
         # Usual metric of between-subject between-voxel correspondence
+        # Making sure ref_ds has most features, if not force feature selection on others
+        nfeatures_all = [sd.nfeatures for sd in ds]
+        nfeatures_bin = [sd.nfeatures > nfeatures for sd in ds]
```

Comment:
since you don't use this "mask" really but just need indices (thus doing np.where below twice), what about
```python
bigger_ds_idxs = [i for i, sd in enumerate(ds) if sd.nfeatures > nfeatures]
```
and then using
```
elif bigger_ds_idxs:
```
in your check and straight those indices where necessary?  imho would make code more concise and thus easier to read.  Choose better name if you like ;)